travis/travis_weblate.py

```diff
@@ -1,166 +1,253 @@
+
+    def _check(self):
+        self.wl_api._check()
+        self.gh_api._check()
+
+    def _apply_patch_odoo(self):
+        """This patch is necessary because the weblate does not check which
+        word and the translated are the same to use it in its percentage of
+        translated"""
+        paths = [os.path.join('openerp', 'tools', 'translate.py'),
+                 os.path.join('odoo', 'tools', 'translate.py')]
+        for path in paths:
+            s_file = os.path.join(self._server_path, path)
+            if not os.path.isfile(s_file):
+                continue
+            cmd = ("sed -i -e \"s/row\\['translation'] = src/"
+                   "row['translation'] = ''/g\" %s" % s_file)
+            print cmd
+            subprocess.call(cmd, shell=True)
+
+    def _get_modules_installed(self):
+        self._installed_modules = []
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            odoo_context.cr.execute("select name from ir_module_module"
+                                    " where state = 'installed'")
+            self._installed_modules = odoo_context.cr.dictfetchall()
+
+    def _generate_odoo_po_files(self, component):
+        generated = False
+        module = []
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            for install in self._installed_modules:
+                if install['name'] in component['filemask']:
+                    module = install['name']
+            if not module:
+                return generated
             print("\n", yellow("Obtaining POT file for %s" % module))
-            i18n_folder = os.path.join(travis_build_dir, module, 'i18n')
-            # # TODO: Add empty es.po files if non exists
-            # source_filename = os.path.join(i18n_folder, module + ".pot")
-            # # Create i18n/ directory if doesn't exist
+            i18n_folder = os.path.join(self._travis_build_dir, module, 'i18n')
             if not os.path.isdir(i18n_folder):
                 os.makedirs(i18n_folder)
-            # with open(source_filename, 'w') as f:
-            #     f.write(odoo_context.get_pot_contents(module))
-
             # Put git add for letting known git which translations to update
-            for po_file_name in os.listdir(i18n_folder):
-                if not po_file_name.endswith('.po'):
+            po_files = glob.glob(os.path.join(i18n_folder, '*.po'))
+            for lang in self._langs:
+                if os.path.isfile(os.path.join(i18n_folder, lang + '.po')):
                     continue
-                lang = os.path.splitext(po_file_name)[0]
-                if langs and lang not in langs:
+                po_content = odoo_context.get_pot_contents(module, lang)
+                with open(os.path.join(i18n_folder, lang + '.po'), 'wb')\
```

Comment:
Remember my previous comment about this matter:
https://github.com/Vauxoo/maintainer-quality-tools/pull/195#discussion_r114590407

<img width="988" alt="screen shot 2017-05-04 at 22 29 48" src="https://cloud.githubusercontent.com/assets/6644187/25732654/c039415c-3117-11e7-9cfb-cc3d883173da.png">